subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/CacheResolveIntegrationTest.groovy

```diff
@@ -112,4 +116,44 @@ project('b') {
         file('a/build/testproject-1.0.jar').assertIsCopyOf(module1.jarFile)
         file('b/build/testproject-1.0.jar').assertIsCopyOf(module2.jarFile)
     }
+
+    def 'dependency cache can be relocated'() {
+        given:
+        def module = ivyHttpRepo.module('group', 'projectA', '1.2').publish()
+
+        and:
+        buildFile << """
+repositories {
+    ivy { url "${ivyHttpRepo.uri}" }
+}
+configurations { compile }
+dependencies { compile 'group:projectA:1.2' }
+task listJars {
+    doLast {
+        assert configurations.compile.collect { it.name } == ['projectA-1.2.jar']
+    }
+}
+"""
+
+        and:
+        module.allowAll()
+
+        and:
+        succeeds('listJars')
+
+        when:
+        server.resetExpectations()
+        renameGradleHome()
+
+        then:
+        succeeds('listJars')
+    }
+
+    def renameGradleHome() {
```

Comment:
:o: Let's call this `relocateCaches` or something like that, so that the intent is clearer.